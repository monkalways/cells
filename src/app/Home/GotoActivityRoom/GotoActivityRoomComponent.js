import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import withStyles from '@material-ui/core/styles/withStyles';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
};

const GotoActivityRoomComponent = ({ classes }) => (
  <Paper className={classes.paper}>
    <Typography component="h1" variant="h5">
      Go to Activity Room?
    </Typography>
    <form className={classes.form}>
      <Button
        type="submit"
        fullWidth
        variant="contained"
        color="primary"
        className={classes.submit}
        disabled
      >
        Go (not ready yet)
      </Button>
    </form>
  </Paper>
);

GotoActivityRoomComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  paper: {
    marginTop: theme.spacing.unit * 8,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 3}px ${theme
      .spacing.unit * 3}px`,
    height: 212,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing.unit,
  },
  submit: {
    marginTop: theme.spacing.unit * 3,
  },
}))(GotoActivityRoomComponent);
