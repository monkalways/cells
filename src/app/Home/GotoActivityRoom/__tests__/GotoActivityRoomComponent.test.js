import React from 'react';
import { shallow } from 'enzyme';

import { Select, Button } from '@material-ui/core';
import { GotoActivityRoomComponent } from '../GotoActivityRoomComponent';

describe('GotoActivityRoomComponent', () => {
  let classes;
  let push;

  beforeEach(() => {
    classes = {
      paper: 'paper',
      form: 'form',
      submit: 'submit',
    };
    push = jest.fn();
  });

  const setup = () => shallow(<GotoActivityRoomComponent classes={classes} history={{ push }} />);

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should set initial state', () => {
    const wrapper = setup();
    expect(wrapper.state().usage).toEqual('');
    expect(wrapper.state().open).toBe(false);
  });

  it('should update open state when select is opened or closed', () => {
    const wrapper = setup();

    wrapper.find(Select).simulate('open');
    expect(wrapper.state().open).toBe(true);

    wrapper.find(Select).simulate('close');
    expect(wrapper.state().open).toBe(false);
  });

  it('should update usage state when select value changes', () => {
    const wrapper = setup();
    const event = {
      target: {
        value: 'phone',
      },
    };
    wrapper.find(Select).simulate('change', event);
    expect(wrapper.state().usage).toEqual('phone');
  });

  it('should handle submit', () => {
    const wrapper = setup();
    wrapper.setState({ usage: 'phone' });
    const preventDefault = jest.fn();
    wrapper.find(Button).simulate('click', { preventDefault });

    expect(preventDefault).toBeCalled();
    expect(push).toBeCalledWith('/activity-rooms/phone/');
  });
});
