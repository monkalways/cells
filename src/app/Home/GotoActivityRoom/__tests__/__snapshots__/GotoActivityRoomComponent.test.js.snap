// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GotoActivityRoomComponent should render 1`] = `
<WithStyles(Paper)
  className="paper"
>
  <WithStyles(Typography)
    component="h1"
    variant="h5"
  >
    Go to Activity Room?
  </WithStyles(Typography)>
  <form
    className="form"
  >
    <WithStyles(FormControl)
      fullWidth={true}
      margin="normal"
      required={true}
    >
      <WithStyles(InputLabel)
        htmlFor="usage"
      >
        Usage
      </WithStyles(InputLabel)>
      <WithStyles(Select)
        inputProps={
          Object {
            "id": "usage",
            "name": "usage",
          }
        }
        onChange={[Function]}
        onClose={[Function]}
        onOpen={[Function]}
        open={false}
        value=""
      >
        <WithStyles(MenuItem)
          value=""
        >
          <em>
            None
          </em>
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          value="bail hearing1"
        >
          Bail Hearing 1
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          value="bail hearing2"
        >
          Bail Hearing 2
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          value="breath test"
        >
          Breath Test
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          value="interview"
        >
          Interview
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          value="medical"
        >
          Medical
        </WithStyles(MenuItem)>
        <WithStyles(MenuItem)
          value="phone"
        >
          Phone
        </WithStyles(MenuItem)>
      </WithStyles(Select)>
    </WithStyles(FormControl)>
    <WithStyles(Button)
      className="submit"
      color="primary"
      fullWidth={true}
      onClick={[Function]}
      type="submit"
      variant="contained"
    >
      Go
    </WithStyles(Button)>
  </form>
</WithStyles(Paper)>
`;
